### PR TITLE
File sink huawei obs

### PR DESCRIPTION
**Pull request by JavaXiaoJun**
huawei obs sink logic

### CODE REVIEW

File: src/main/java/org/apache/pulsar/io/sftp/sink/MessageToParquetFileOBSWriter.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+ */
+@Slf4j
+public class MessageToParquetFileOBSWriter implements MessageOBSWriter<byte[]> {
+
+    @Override
+    public void writeToStorage(Record<byte[]> record, OBSSink obsSink, ObsConfiguration conf) {
+        OBSSinkConfig sinkConfig = obsSink.getOBSSinkConfig();
+        String outDirectory = sinkConfig.getOutDirectory();
+        //TODO
+        String parquetFileName = "file-" + new Date().getTime() + ".parquet";
```
Comment (liudezhi2098) [inline src/main/java/org/apache/pulsar/io/sftp/sink/MessageToParquetFileOBSWriter.java:57]: If it is Debezium data , we can prefix it with the table name

File: src/main/java/org/apache/pulsar/io/sftp/sink/MessageToParquetFileOBSWriter.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        // The file is temporarily stored in the server's directory and will be deleted after uploading obs
+        String tempParquetFilePath = "/tmp" + outDirectory + "/" + parquetFileName;
```
Comment (liudezhi2098) [inline src/main/java/org/apache/pulsar/io/sftp/sink/MessageToParquetFileOBSWriter.java:59]: May need to be careful if inside a pod

File: src/main/java/org/apache/pulsar/io/sftp/sink/MessageToRawFileOBSWriter.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                    : outDirectory.replaceFirst("/", "") + "/" + name;
+            String originalMD5 = msg.getProperty(Constants.FILE_MD5);
+            ObsClient obsClient = HWObsUtil.getObsClient(sinkConfig.getAccessKey(), sinkConfig.getSecretKey(),
+                    sinkConfig.getSecurityToken(), conf);
+            PutObjectRequest request = new PutObjectRequest();
+            request.setInput(new ByteArrayInputStream(contents));
+            request.setBucketName(sinkConfig.getBucket());
+            request.setObjectKey(fileName);
+            request.setExpires(sinkConfig.getExpires());
+            PutObjectResult result = obsClient.putObject(request);
```
Comment (liudezhi2098) [inline src/main/java/org/apache/pulsar/io/sftp/sink/MessageToRawFileOBSWriter.java:68]: You can first determine whether the file exists and whether the MD5 is consistent. If it is consistent, you do not need to upload it again.

File: src/main/java/org/apache/pulsar/io/sftp/sink/MessageToRawFileOBSWriter.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            log.info("Put Object to OBS success , Path : " + result.getObjectUrl());
+
+            String currentMD5 = FileUtil.getFileMD5(contents);
```
Comment (liudezhi2098) [inline src/main/java/org/apache/pulsar/io/sftp/sink/MessageToRawFileOBSWriter.java:71]: We finally get the md5 value of the file in obs directly

File: src/main/java/org/apache/pulsar/io/sftp/sink/MessageToParquetFileOBSWriter.java (hunk 1 above)
Comment (JavaXiaoJun) [reply src/main/java/org/apache/pulsar/io/sftp/sink/MessageToParquetFileOBSWriter.java:57]: got it

File: src/main/java/org/apache/pulsar/io/sftp/sink/MessageToRawFileOBSWriter.java (hunk 4 above)
Comment (JavaXiaoJun) [reply src/main/java/org/apache/pulsar/io/sftp/sink/MessageToRawFileOBSWriter.java:71]: Is this to make sure the file doesn't change during upload to obs?

File: src/main/java/org/apache/pulsar/io/sftp/sink/MessageToParquetFileOBSWriter.java (hunk 2 above)
Comment (JavaXiaoJun) [reply src/main/java/org/apache/pulsar/io/sftp/sink/MessageToParquetFileOBSWriter.java:59]: Can the temporary path of this file be specified by the user?

**Pull request merged**